backend/app/store/service/restricted_words.go

```diff
@@ -0,0 +1,95 @@
+package service
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+type Lister interface {
+	List(siteID string) (restricted []string, err error)
+}
+
+type StaticLister struct {
+	Words map[string][]string
+}
+
+func (l *StaticLister) List(siteID string) (restricted []string, err error) {
+	restricted, exists := l.Words[siteID]
+	if !exists {
+		return restricted, fmt.Errorf("no restricted words configured for site '%s'", siteID)
+	}
+	return
+}
+
+type Matcher struct {
+	lister Lister
+	data   map[string]restrictedWordsSet
+}
+
+type restrictedWordsSet struct {
+	restricted map[string]bool
+}
+
+func NewMatcher(lister Lister) *Matcher {
+	return &Matcher{lister, make(map[string]restrictedWordsSet)}
+}
+
+func (m *Matcher) Match(siteID string, text string) bool {
+	tokens := m.tokenize(text)
+
+	data, exists := m.data[siteID]
+	if !exists {
+		words, err := m.lister.List(siteID)
+		if err != nil {
+			fmt.Printf("failed to get restricted words for site %s: %v", siteID, err)
+			return false
+		}
+		restricted := make(map[string]bool)
+		for _, w := range words {
+			restricted[strings.ToLower(w)] = true
+		}
+		data = restrictedWordsSet{restricted}
+		m.data[siteID] = data
+	}
+
+	for _, token := range tokens {
+		_, present := data.restricted[token]
+		if present {
+			return true
+		}
+	}
+	return false
+}
+
+func (_ *Matcher) tokenize(text string) []string {
```

Comment:
is it even needed? I think string pkg has smth to split with whitespaces

Reply:
This one also strips punctuation, spaces, tabs, emojis. See tests. For me it seems strange to have tokenizer that leaves such trash. It would be harder to match these tokens later.

Reply:
ok, makes sense